src/Data/Connection/MenuItemConnectionResolver.php

```diff
@@ -70,30 +84,44 @@ private static function get_menu_items( $source, array $args ) {
 	 * them with various filters, etc and returns an array of $query_args to be used in the
 	 * \WP_Query call
 	 *
-	 * @param mixed       $source  The query source being passed down to the resolver
-	 * @param array       $args    The arguments that were provided to the query
-	 * @param AppContext  $context Object containing app context that gets passed down the resolve
-	 *                             tree
-	 * @param ResolveInfo $info    Info about fields passed down the resolve tree
-	 *
 	 * @return array
 	 * @throws \Exception
 	 * @since  0.0.30
 	 */
-	public static function get_query_args( $source, array $args, AppContext $context, ResolveInfo $info ) {
+	public function get_query_args() {
+
+
+		/**
+		 * Filter the $this->args to allow folks to customize query generation programmatically
+		 *
+		 * @param array       $args       The inputArgs on the field
+		 * @param mixed       $source     The source that's passed down the GraphQL queries
+		 * @param AppContext  $context    The AppContext passed down the GraphQL tree
+		 * @param ResolveInfo $info       The ResolveInfo passed down the GraphQL tree
+		 */
+		$args = apply_filters( 'graphql_menu_item_connection_args', $this->args, $this->source, $this->context, $this->info );
 
 		// Prevent the query from matching anything by default.
 		$query_args = [
 			'post_type' => 'nav_menu_item',
-			'post__in'  => array( 0 ),
```

Comment:
apparently at one point this was needed to prevent some edge case. The `post__in` gets defined later, but only in certain cases, and I guess for certain cases this was needed as the default. 

I don't recall specifics now, but it seemed to be causing me issues after refactoring other internal.